### ISSUE TITLE
Get rid of the polling interval

### ISSUE THREAD

**Issue by miracle2k** [seomoz/qless-py#49]
Browsing the code, it seems that qless already gets new jobs via pubsub by redis. Is there a reason why it uses the "sleep interval" pattern currently? It would seem trivial to just let it handle any new incoming job immediately.

**Comment by miracle2k** [comment]
I think I made a mistake; The "pop" command is not actually blocking. Still, this would be nice to have.

**Comment by dlecocq** [comment]
That's correct -- there is no blocking pop. It is a bit of a wart in that it it puts a lower bound on the latency between job put and pop. The overhead of polling in this case is actually pretty negligible, however -- if there are enough jobs to keep workers saturated, every `pop` gets a job, and so no work is wasted. Upon completing a job, it also attempts to pop the next. If there aren't enough jobs to keep workers saturated, then it's in some sense safe to waste the overhead of a non-job-returning `pop` because the cluster's somewhat idle.

**Comment by miracle2k** [comment]
Makes sense. Is it technically possible to implement a blocking pop with Lua in Redis?
